mate-panel/drawer.c

```diff
@@ -587,7 +587,10 @@ panel_drawer_prepare (const char  *drawer_id,
     g_settings_set_boolean (settings, PANEL_OBJECT_USE_CUSTOM_ICON_KEY, use_custom_icon);
 
     if (custom_icon) {
-        g_settings_set_string (settings, PANEL_OBJECT_CUSTOM_ICON_KEY, custom_icon);
+	gchar *icon_name;
+	icon_name = g_icon_to_string(custom_icon);
+        g_settings_set_string (settings, PANEL_OBJECT_CUSTOM_ICON_KEY, icon_name);
+	g_free(icon_name);
```

Comment:
Something wrong with indentation, also there should be space after function name like it is done with `g_settings_set_string`.